### PR TITLE
otter: bringup

### PR DESCRIPTION
**Pull request by amartinz**

### CODE REVIEW

**Comment by amartinz** [comment]
@z3ntu Commit https://github.com/sc7280-mainline/linux/pull/2/commits/8b7f9ea6ea761eaae7dee51830c3992c2f3d3719 is to be in sync with https://gitlab.com/sdm845-mainline/linux/-/commit/3a2247fe161ad4dec262f9eed1ea9ccfef65425a but can be dropped, as audio does not work yet anyways.

**Comment by z3ntu** [comment]
I've just pushed sc7280-6.10-wip, could you rebase on that? Also what's missing for this to not be a Draft?

**Comment by amartinz** [comment]
> I've just pushed sc7280-6.10-wip, could you rebase on that?

Sure, will do!

> Also what's missing for this to not be a Draft?

I need to do a fresh install of pmOS with all the local adjustments removed and test it.
Right now it was building the sdm845 kernel instead with these changes added on top.

Also i wanted to get feedback (forgot to ask) about the defconfig changes.
If you prefer to have a defconfig for all devices (like we do with sdm845) or device specific ones.

**Comment by z3ntu** [comment]
I'm good with a shared sc7280_defconfig, should only be a small handful of options for the different devices anyways, the rest is SoC-specific so shared.

File: arch/arm64/configs/sc7280_defconfig [hunk 1]
```diff
@@ -361,11 +361,9 @@ CONFIG_SND_USB_AUDIO=y
 CONFIG_SND_SOC=y
 CONFIG_SND_SOC_QCOM=y
 CONFIG_SND_SOC_SM8250=y
+CONFIG_SND_SOC_SC7280=y
```
Comment (z3ntu) [inline arch/arm64/configs/sc7280_defconfig:366]: Qualcomm people are adding qcm6490 sound to sc8280xp driver, so I don't think we're going to use sc7280 driver at all? https://lore.kernel.org/all/20240524035350.3118981-3-quic_mohs@quicinc.com/

Btw the sm8250 machine driver that's already enabled is kind of temporarily used on FP5, functionality-wise they do the same thing for now, I'll sort that out later to move to sc8280xp driver

File: arch/arm64/configs/sc7280_defconfig [hunk 2]
```diff
@@ -428,6 +426,7 @@ CONFIG_QCOM_CLK_RPMH=y
 CONFIG_SC_CAMCC_7280=y
 CONFIG_SC_DISPCC_7280=y
 CONFIG_SC_GPUCC_7280=y
+CONFIG_SC_LPASSCC_7280=y
```
Comment (z3ntu) [inline arch/arm64/configs/sc7280_defconfig:431]: I don't think we're going to use that one, this driver touches registers that are owned by adsp on our devices, only on Chromebooks this is useful

File: drivers/gpu/drm/panel/panel-shift-sh8804b.c [hunk 3]
```diff
@@ -0,0 +1,1189 @@
...
+
+	ret = mipi_dsi_detach(dsi);
+	if (ret < 0)
+		dev_err(&dsi->dev, "Failed to detach from DSI host: %d\n", ret);
+
+	drm_panel_remove(&ctx->panel);
+}
+
+static const struct of_device_id shift_sh8804b_of_match[] = {
+	{ .compatible = "shift,sh8804b" }, // FIXME
```
Comment (z3ntu) [inline drivers/gpu/drm/panel/panel-shift-sh8804b.c:1176]: Maybe you can find a nice model name or something from your display vendor?

E.g. for Fairphone 5 I have a pdf for the company Boe with the "Model Name" BF065GBM-TK0-7DP0 which is exactly the combo of driver IC (rm692e5) plus everything else that's (I assume) specific to Fairphone 5.

Otherwise probably upstream is happy enough if you make it something else, see e.g. Fairphone 5 panel upstream (to be clear, that doesn't follow what I was writing, see Fairphone 4 panel there: `panel-himax-hx83112a.c`)

File: arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts [hunk 4]
```diff
@@ -533,6 +533,54 @@
...
+	status = "okay";
+
+	panel@0 {
+		compatible = "shift,sh8804b";
+		reg = <0>;
+
+		reset-gpios = <&tlmm 44 GPIO_ACTIVE_LOW>;
+
+		vci-supply = <&vreg_l13c>;
+		vddio-supply = <&vreg_l12c>;
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts:568]: No other supplies? Double check the schematics so that we have complete bindings

File: arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts [hunk 5]
```diff
@@ -533,6 +533,54 @@
...
+	data-lanes = <0 1 2 3>;
+	remote-endpoint = <&panel_in>;
+};
+
+&mdss_dsi_phy {
+	vdds-supply = <&vreg_l10c>;
+	status = "okay";
+};
+
+&mdss_dp {
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts:592]: mdss_dp < mdss_dsi, please fix sorting :)

File: arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts [hunk 6]
```diff
@@ -533,6 +533,54 @@
...
+	status = "okay";
+};
+
+&mdss_dp_out {
+	data-lanes = <0 1>;
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts:597]: fyi we can hopefully soon enable 4 lane DP here, there's some patches on the mailing list to enable some bits driver-side

File: arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts [hunk 7]
```diff
@@ -881,6 +898,66 @@
...
+				function = "gpio";
+			};
+
+			config {
+				pins = "gpio105", "gpio81";
+				drive-strength = <2>;
+				bias-disable;
+			};
+		};
+	};
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts:973]: Can you at least collapse these nodes to e.g.

```c
pmx-ts-release-state {
	pins = "gpio105", "gpio81";
	drive-strength = <2>;
	bias-disable;
}
```
So they also pass dtbs_check when we run it

File: arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts [hunk 8]
```diff
@@ -523,7 +523,24 @@
...
 	status = "okay";
 
-	/* focaltech FT3658U @ 38 */
+	touchscreen@38 {
+		compatible = "focaltech,fts";
+		reg = <0x38>;
+
+		/* FIXME!!!! */
+		interrupt-parent = <&tlmm>;
+		interrupts = <81 0x2008>;
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts:532]: I guess with FIXME you mean the 0x2008? Please specify ;) Also since it's not going upstream like this anyways imo if it works, remove the fixme

File: arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/qcm6490-shift-otter.dts:1]: `arm64: dts: qcom: qcm6490-shift-otter: flash led fix` commit message

Reword this to "enable flash led" or something? The current contents doesn't 'fix' anything, it just configures and enables it

File: arch/arm64/configs/sc7280_defconfig [hunk 9]
```diff
@@ -348,6 +349,7 @@ CONFIG_DRM_MSM=y
 # CONFIG_DRM_MSM_DSI_10NM_PHY is not set
 # CONFIG_DRM_MSM_HDMI is not set
 CONFIG_DRM_PANEL_RAYDIUM_RM692E5=y
+CONFIG_DRM_PANEL_SHIFT_SH8804B=m
```
Comment (z3ntu) [inline arch/arm64/configs/sc7280_defconfig:352]: you don't want to have those built-in? Also FOCALTECH_FT3658U? This defconfig is imo only for development anyways so making everything built-in simplifies development

**Comment by amartinz** [comment]
I have talked with Caleb and it makes sense to not merge this and really go for upstream first.

Like, if the downstream touch driver ever gets merged, surely noone will bother trying to rewrite it for upstream as "it already works".

Thanks a lot for the review and will try to adjust so these patches are upstreamable instead and only send early PRs, once patches are applied upstream.

What do you think?

File: drivers/gpu/drm/panel/panel-shift-sh8804b.c (hunk 3 above)
Comment (amartinz) [reply drivers/gpu/drm/panel/panel-shift-sh8804b.c:1176]: This would be `TA067FVMK01-08` from `TIANMA` then, also using `rm692e5`, but they also made adjustments specific for SP8.

**Comment by z3ntu** [comment]
> Like, if the downstream touch driver ever gets merged, surely noone will bother trying to rewrite it for upstream as "it already works".

It's definitely currently the situation for FP4 + FP5 for me, but I honestly also haven't really seen any other code contributions for those devices so I'm not sure how much it matters really.

> Thanks a lot for the review and will try to adjust so these patches are upstreamable instead and only send early PRs, once patches are applied upstream.
> 
> What do you think?

Also fine for me, but I'm also happy this being a staging area for patches that do work but haven't landed upstream yet - or simple standalone stuff like touchscreen drivers and whatnot.

**Comment by amartinz** [comment]
Closing in favor of a new PR.
